frontend/src/components/DropdownButton.tsx

```diff
@@ -0,0 +1,100 @@
+import { MouseEvent, MouseEventHandler, ReactNode, useState } from 'react'
+import { Button, ButtonProps, MenuProps } from '@mui/material'
+import Menu from '@mui/material/Menu'
+import clsx from 'clsx'
+
+import downArrow from '@/assets/svgs/fa-caret-down.svg'
+
+import './DropdownButton.css'
+
+const defaultButtonStyles = {
+  display: 'flex',
+  gap: 0,
+}
+
+interface ButtonDropdownProps extends ButtonProps {
+  id: string
+  onClick?: MouseEventHandler
+  dropdownContent?: ReactNode
+  children?: ReactNode
+  showArrow?: boolean
+  arrowProps?: Record<string, any>
+  openClassName?: string
+  menuClassName?: string
+  menuProps?: Partial<MenuProps>
+  [key: string]: any
+}
+
+export function DropdownButton({
```

Comment:
This component renders a button, that when clicked will show a Menu with any custom component as the content.